contracts/nexus-gateway/src/contract/migrations/v1_0_0.rs

```diff
@@ -0,0 +1,126 @@
+use axelar_wasm_std::address;
+use axelar_wasm_std::error::ContractError;
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{Addr, DepsMut};
+use cw_storage_plus::Item;
+
+use crate::contract::{CONTRACT_NAME, CONTRACT_VERSION};
+use crate::msg::MigrateMsg;
+use crate::state;
+
+const BASE_VERSION: &str = "1.0.0";
+const CONFIG: Item<Config> = Item::new("config");
+
+pub fn migrate(deps: &mut DepsMut, msg: MigrateMsg) -> Result<(), ContractError> {
+    cw2::assert_contract_version(deps.storage, CONTRACT_NAME, BASE_VERSION)?;
+
+    let axelarnet_gateway = address::validate_cosmwasm_address(deps.api, &msg.axelarnet_gateway)?;
+    migrate_config(deps, axelarnet_gateway)?;
+
+    cw2::set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
```

Comment:
can we set this in the migrate entry point, instead of here? So that way we are consistent with contracts that don't actually have any migration logic, but just need to migrate the code

Reply:
I don't see how is it consistent https://github.com/axelarnetwork/axelar-amplifier/blob/2acf237d4c8cbe20e844436c2b94259408020597/contracts/voting-verifier/src/contract/migrations/v0_5_0.rs#L26.

Reply:
we need to come up with a clear migration strategy and then set up consistent rules from there. Right now it's way to easy to release versions with incompatible upgrade paths